adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/geoutils/GeospatialUtilsImpl.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright (c) 2020. MobilityData IO.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.geoutils;
+
+import org.locationtech.spatial4j.distance.DistanceUtils;
+import org.mobilitydata.gtfsvalidator.usecase.utils.GeospatialUtils;
+
+/**
+ * Utility class to carry out operations related to geospatial data
+ */
+public class GeospatialUtilsImpl implements GeospatialUtils {
+    private static GeospatialUtilsImpl GEO_UTILS = null;
+
+    private GeospatialUtilsImpl() {
+    }
+
+    /**
+     * Implement singleton pattern
+     *
+     * @return a unique instance of {@code GeospatialUtilsImpl}
+     */
+    public static GeospatialUtilsImpl getInstance() {
+        if (GEO_UTILS == null) {
+            GEO_UTILS = new GeospatialUtilsImpl();
+        }
+        return GEO_UTILS;
+    }
+
+    /**
```

Comment:
There is an alternate implementation of this in https://github.com/MobilityData/gtfs-validator/pull/302/files#diff-2a4deacfd60d7ef9315e71a0b66324beR53 that uses the `DistanceCalculator` class - https://locationtech.github.io/spatial4j/apidocs/com/spatial4j/core/distance/DistanceCalculator.html#distance(com.spatial4j.core.shape.Point,%20com.spatial4j.core.shape.Point).

Was the DistanceUtils class used for a reason here?

Reply:
No other reason than not remembering the other PR and feeling like it is less code.

Reply:
I'll ask @lionel-nj to backport his implementation in this PR.

Reply:
@fabrice-v, see f936c7f7d6f190709b67687a4f9384faa7d08ae9.